applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_implicit_dynamic_solver.py

```diff
@@ -58,12 +58,25 @@ def AddDofs(self):
         self._add_dynamic_dofs()
         self.print_on_rank_zero("::[ImplicitMechanicalSolver]:: ", "DOF's ADDED")
 
+    def InitializeSolutionStep(self):
+        # Using the base InitializeSolutionStep
+        super(ImplicitMechanicalSolver, self).InitializeSolutionStep()
+
+        # Some pre-processes may affect the system of equations, we rebuild the equation ids
+        if self.main_model_part.ProcessInfo[KratosMultiphysics.STEP] == 1:
+            # Resetting the global equations ids
+            self.get_builder_and_solver().SetUpSystem(self.GetComputingModelPart())
```

Comment:
Why is this needed here?

This has the potential to confuse and also break things

=> cannot you do this in your process? (there you also have a B&S)

Reply:
I cannot access to the builder and solver in the process. If you use the process the EquationId are assigned, you need to reset it. Sincerely I don't know how to fix it